### PR TITLE
refactor(tx-pool): add enum `InvalidKind` to `mark_invalid`

### LINKED ISSUE #12405: Convey more information about an invalid transaction
### Describe the feature

currently we have

https://github.com/paradigmxyz/reth/blob/9f6f63d45a1a5b9edf3087a9210e6a9170e4c1ec/crates/transaction-pool/src/traits.rs#L755-L760

which is usually invoked for several reasons:

* exceeds gas
https://github.com/paradigmxyz/reth/blob/9f6f63d45a1a5b9edf3087a9210e6a9170e4c1ec/crates/ethereum/payload/src/lib.rs#L226-L229

* invalid transaction
https://github.com/paradigmxyz/reth/blob/9f6f63d45a1a5b9edf3087a9210e6a9170e4c1ec/crates/ethereum/payload/src/lib.rs#L268-L271

we can add this info as additional context via an enum to mark_invalid

## TODO
* introduce an enum like `InvalidKind { ExceedsGasLimit, InvalidTransaction(Err),... }`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lean-apple**
Closes https://github.com/paradigmxyz/reth/issues/12405.

### CODE REVIEW

File: crates/transaction-pool/src/traits.rs [hunk 1]
```diff
@@ -753,6 +753,33 @@ impl fmt::Display for CanonicalStateUpdate<'_> {
...
+
+/// Specifies which type of gas limit was exceeded
+#[derive(Debug, Clone, Copy, derive_more::Display)]
+pub enum GasLimitKind {
+    /// Block gas limit exceeded
+    Block,
+    /// Blob gas limit exceeded
+    Blob,
+}
+
```
Comment (lean-apple) [inline crates/transaction-pool/src/traits.rs:782]: I created this enum as suggested in the issue's description, yet why not reusing the [PoolErrorKind](https://github.com/paradigmxyz/reth/blob/main/crates/transaction-pool/src/error.rs#L41) enum instead @mattsse and adding some new variants ?

Comment (onbjerg) [reply crates/transaction-pool/src/traits.rs:782]: I actually think `InvalidPoolTransactionError `https://github.com/paradigmxyz/reth/blob/404f8f877842efa887a1b39e78de2eb6ec97289d/crates/transaction-pool/src/error.rs#L189 might be better?

File: crates/ethereum/payload/src/lib.rs [hunk 2]
```diff
@@ -250,7 +257,14 @@ where
...
                 // for regular transactions above.
                 trace!(target: "payload_builder", tx=?tx.hash, ?sum_blob_gas_used, ?tx_blob_gas, "skipping blob transaction because it would exceed the max data gas per block");
-                best_txs.mark_invalid(&pool_tx);
+                best_txs.mark_invalid(
+                    &pool_tx,
+                    InvalidKind::ExceedsGasLimit {
+                        kind: GasLimitKind::Blob,
+                        required: tx_blob_gas,
+                        limit: MAX_DATA_GAS_PER_BLOCK,
+                    },
```
Comment (mattsse) [inline crates/ethereum/payload/src/lib.rs:266]: I'd like to have a few initialize fns on InvalidKind for both blob, regular

File: crates/transaction-pool/src/traits.rs [hunk 3]
```diff
@@ -753,6 +753,33 @@ impl fmt::Display for CanonicalStateUpdate<'_> {
     }
 }
 
+/// Helper error type representing different reasons why a transaction might fail during
+/// processing in the transaction pool.
+#[derive(Debug, derive_more::Display, derive_more::Error)]
+pub enum InvalidKind {
```
Comment (mattsse) [inline crates/transaction-pool/src/traits.rs:759]: I think there might be other reasons that aren't necessarily consensus-related, so I think we can also add an `Other` variant here

**Review by mattsse — CHANGES_REQUESTED** [review]
cool, some nits

File: crates/transaction-pool/src/traits.rs (hunk 1 above)
Comment (lean-apple) [reply crates/transaction-pool/src/traits.rs]: I replaced the created enum by using directly `InvalidPoolTransactionError`

File: crates/transaction-pool/src/traits.rs (hunk 3 above)
Comment (lean-apple) [reply crates/transaction-pool/src/traits.rs]: `InvalidPoolTransactionError` has already an `Other`variant, yet I thought 
```rust    
InvalidPoolTransactionError::Consensus(InvalidTransactionError::TxTypeNotSupported),
``` could be adapted

**Review by mattsse — APPROVED** [review]
great!

ty

**Pull request merged**